addons/flannel/0.21.5/install.sh

```diff
@@ -270,9 +270,20 @@ function remove_weave() {
 
     # all nodes
     ip link delete weave
+
     rm -rf /var/lib/weave
     rm -rf /etc/cni/net.d/*weave*
     rm -rf /opt/cni/bin/weave*
+
+    logStep "Verifying Weave removal"
+    # Check if any Weave-related resources are still present
+    weave_resources=$(kubectl get all --all-namespaces | grep weave || true)
+
+    if [ -n "$weave_resources" ]; then
+        logWarn "Weave resources are still present. Force deleting Weave resources..."
+        kubectl -n kube-system delete $(kubectl get all -n kube-system | grep weave | awk '{print $1}') --force --grace-period=0
+    fi
+    logSuccess "Weave has been successfully removed."
```

Comment:
if you delete with force and grace-period=0 does kubectl wait for pods to be terminated and removed? if not can you log this with confidence?